tests/test_views.py

```diff
@@ -37,8 +39,10 @@ def test_get_auth_success(self):
         }
         url = reverse('oidc_authentication_callback')
         request = self.factory.get(url, get_data)
-        request.session = {
-            'oidc_state': 'example_state'
+        client = Client()
```

Comment:
I'm curious why it was necessary to copy a session from ``Client()``. I have wrestled with sessions in a test environment however, so I'm not too surprised.

Reply:
The requests object created by the factory does not have a session attribute! I think there are two possible ways to get a session attribute:
1. Like above, use the existing session object from the client. As the `Client` class is based on `RequestFactory`, we could use a `Client` directory instead of the `RequestFactory`.

2. We could do it similar like here: https://github.com/akatsoulas/mozilla-django-oidc/blob/b18fbd00240f89f02528363da2af61f881e28d26/tests/test_utils.py#L60
But this would require executing the session in each test method as the URL/data parameters do change in every test method compared to the test_utils.py where the request is always the same.

Reply:
That sounds familiar, and the [Django docs](https://docs.djangoproject.com/en/3.0/topics/testing/advanced/#django.test.RequestFactory) confirm it. A little weirdness is expected when testing authentication code.